GLA Request Review - Tracking events

puntope opened the pull request:
### Changes proposed in this Pull Request:


Closes part of #1165 

This PR will add support for tracking events in GLA Request Review Module

- Open the modal on Request Review click. `gla_modal_open`
- Close the modal. `gla_modal_closed`  
- Check/Uncheck the Issues solved checkbox.  `gla_request_review_issues_solved_checkbox_click` 
- Expand/Collapse the issue list. `gla_request_review_issue_list_toggle_click`
- Click on the Request Review button inside the modal. `gla_request_review`

### Screenshots:

https://user-images.githubusercontent.com/5908855/155144219-08bbc493-4a5d-4f5e-b816-11a95e13b4cb.mov


### Detailed test instructions:

### Instructions before the tests

1. Be sure you have an account connected and at least one issue. You can create one issue in the `wp_gla_merchant_issues` table. 
2. Enable tracking in Woo -> Settings -> Advanced -> Woocommere.com -> Enable tracking
<img width="1539" alt="Screenshot 2022-02-22 at 15 34 05" src="https://user-images.githubusercontent.com/5908855/155144790-de4e555f-82fb-4cf8-ad5e-c1887fef655e.png">

3. Open your console and execute this `localStorage.setItem('debug', 'wc-admin*')`
4. Enable verbose mode in console. 
<img width="612" alt="Screenshot 2022-02-22 at 15 35 02" src="https://user-images.githubusercontent.com/5908855/155144760-6785b42e-31ac-4bc9-ad47-400f216b01a3.png">
5. Go to WooCommerce -> GLA -> Product feed `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed` 

### Instructions for Events

**Modal Open**

1. Click on the Request Review button. 
2. See how an event `gla_modal_open` appears with the `context` prop as `request_review` 

**Issue expand/collapse**
1. After opening the modal, click on `+1 more issue(s)`. 
2. See how an event `gla_request_review_issue_list_toggle_click` appears with the `action` prop as `expand` 
3. Click now again in the same button, now named `Show less`
4. See how an event `gla_request_review_issue_list_toggle_click` appears with the `action` prop as `collapse` 

**issue solved Checkbox**
1. After opening the modal, click on `I have resolved all the issue(s) listed above.` 
2. See how an event `gla_request_review_issues_solved_checkbox_click` appears with the `action` prop as `check`
3. Click again on the checkbox for disabling it
4. See how an event `gla_request_review_issues_solved_checkbox_click` appears with the `action` prop as `uncheck`

**Close modal**
1. After opening the modal, click on cancel
2. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `cancel`
3. After opening the modal again, click on the X in the corner
4. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `dismiss`
5. After opening the modal again, click outside the modal
6. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `dismiss`

**Review request**
1. After opening the modal, click o. the checkbox and then on the Button Request Review
2. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `request_review`
3. See how an event `gla_request_review` appears 

### Changelog entry

> Add Tracking in Request Review Module

puntope committed:
Tracking events

puntope committed:
Improve tracking props

puntope committed:
Merge branch 'feature/1165-gmc-account-review-request' into add/1165-tracking

# Conflicts:
#	src/API/Site/Controllers/MerchantCenter/AccountReviewController.php

eason9487 commented on js/src/product-feed/review-request/index.js:
```diff
@@ -29,18 +30,36 @@ const ReviewRequest = ( { account = {} } ) => {
 
+	const handleNoticeClick = () => {
+		setModalActive( true );
+		recordEvent( 'gla_modal_open', { context: 'request_review' } );
```
💅 What about giving a constant variable for `'request_review'`?

eason9487 commented on js/src/product-feed/review-request/index.js:
```diff
@@ -29,18 +30,36 @@ const ReviewRequest = ( { account = {} } ) => {
+	};
+
+	const handleReviewRequest = () => {
+		handleModalClose( 'request_review' );
```
💅 What about using another `action` string to make it a little differentiated from `context`? Such as `confirm` or `confirm_request_review`.

![image](https://user-images.githubusercontent.com/17420811/156698994-373a74b6-793d-40c3-b68b-072bffec033c.png)

eason9487 reviewed (commented):
All tracking events are triggered as expected. Left some minor questions. And I think it would be better to add these new tracking names to *src/Tracking/README.md*:
- request_review
- request_review_issue_list_toggle_click
- request_review_issues_solved_checkbox_click

puntope committed:
Update naming and add constants

puntope committed:
Update naming and TReadme

puntope commented:
Howdy @eason9487 

I pushed some commits based on your suggestions:

- Updated Tracking README. (Added the new functions and fixed some typos in other 2, also my IDE changed the format of some spaces)
- Adding some constant regarding `request-review` and also moving `product-feed/constants.js` file (with only 1 constant) inside `src/constants.js `
- Change `cancel` action in Modal Close to `maybe-later` to be consistent with the Modal Close current definition
- Change `request-review` action innModal Close to `confirm-request-review` 

Could you take another look? 🙇

eason9487 reviewed (approved):
Thanks for the adjustments! LGTM. A little thing about formatting could be considered below. Approved in advance. 👍 

> my IDE changed the format of some spaces

The file was auto-formatted by prettier in #876. So you could run it again to make them have the same spaces in this repo.
```bash
./node_modules/prettier/bin-prettier.js --write ./src/Tracking/README.md
```

puntope committed:
Run prettier manually
